### ISSUE TITLE
resin deply can't handle hyphens in application names

### ISSUE THREAD

**Issue by zut** [resin-io/resin-cli#913]
- **resin-cli version:**
```
> resin version
7.4.0
```
- **Operating system and architecture:**
OS X 10.13.5, x86_64

Easy to reproduce:
```
> resin deploy "company/foo-bar"
ResinApplicationNotFound: Application not found: company/foo

If you need help, don't hesitate in contacting us at:

  GitHub: https://github.com/resin-io/resin-cli/issues/new
  Forums: https://forums.resin.io

```

**Comment by pimterry** [comment]
Great bug report, thanks! I think this'll actually be fixed by some of the changes in https://github.com/resin-io/resin-cli/pull/895, which should be landing very soon.

Otherwise in the meantime you should be able to use the application's unique id instead (it's in the dashboard URL or the output of `resin apps`) as a workaround. I'll leave this open for now until that other PR is merged fixing this.

**Comment by pimterry** [comment]
This has now been resolved by #895, released as 7.8.0. Thanks for the report! Let us know if you have any other trouble.
